lib/public/json_converter.dart

```diff
@@ -72,9 +72,42 @@ class _DateTimeJsonConverter implements JsonConverter<DateTime, String> {
   DateTime fromJson(String value) => DateTime.parse(value);
 }
 
+class _DurationJsonConverter implements JsonConverter<Duration, String> {
+  const _DurationJsonConverter();
+
+  @override
+  String toJson(Duration value) => value.toString();
+
+  @override
+  Duration fromJson(String value) => value.parseDuration();
+}
+
 /// Registrant for adding new converters used by [fromJson] and [toJson]
 ///
 /// see also on how to register a converter [JsonConverter]
 final JsonConverterRegistrant jsonConverterRegistrant = JsonConverterRegistrant()
   ..register(const _UriJsonConverter())
-  ..register(const _DateTimeJsonConverter());
+  ..register(const _DateTimeJsonConverter())
+  ..register(const _DurationJsonConverter());
+
+extension on String {
+  Duration parseDuration() {
```

Comment:
I think the logic is not entirely correct, because the format of duration in string it's normally like this

[HH:][MM:]SS[.mmmμμμ]

But also the time can be saved an integer, in milliseconds or seconds (most frequent cases).
Let's have a discussion about this